src/Chapter4.hs

```diff
@@ -293,7 +301,18 @@ values and apply them to the type level?
 -}
 instance Functor (Secret e) where
     fmap :: (a -> b) -> Secret e a -> Secret e b
-    fmap = error "fmap for Box: not implemented!"
+    fmap f (Reward a) = Reward (f a)
+    fmap _ (Trap e) = Trap e
+
+{- 
+@Question how to apply functor if my data is defined as
+  data Secret e a
+    = Reward e
+    | Trap a
+    deriving (Show, Eq)
+  
+  I may not have access to modules
+-}
```

Comment:
I'm not sure what do you mean by having access to modules. You still can implement the `Functor` instance for this new different `Secret` data type, but it will map inside the `Trap` constructor instead, because the `a` type variable is last.